std/http/cookie.ts

```diff
@@ -32,11 +32,14 @@ export interface Cookie {
 
 export type SameSite = "Strict" | "Lax" | "None";
 
+const FIELD_CONTENT_REGEXP = /^[\u0009\u0020-\u007e\u0080-\u00ff]+$/;
```

Comment:
Where is this character set from?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie gives valid character sets for cookie names and values, neither of which are this. I assume `domain` and `path` are encoded like they are in URLs.

Reply:
@nayeemrmn what about other browser like [internet explorer]() and [Safari - Apple](https://www.apple.com/safari/) ?

Reply:
> @nayeemrmn what about other browser like internet explorer and [Safari - Apple](https://www.apple.com/safari/) ?

What about them? Is the MDN page I linked not consistent with the relevant spec?

Reply:
@nayeemrmn So we should remove validation property from path and domain ?

Reply:
@getspooky Could you use the character set specified on MDN (or the spec) for each of the values being checked? I don't know where you got this regexp from, but it doesn't look right.

Reply:
@lucacasonato https://github.com/pillarjs/cookies/blob/master/index.js

Reply:
@lucacasonato  any feedback ?

Reply:
From MDN:

> - A `<cookie-name>` can be any US-ASCII characters, except control characters, spaces, or tabs. It also must not contain a separator character like the following: `( ) < > @ , ; : \ " / [ ] ? = { }`.
> - A `<cookie-value>` can optionally be wrapped in double quotes and include any US-ASCII characters excluding control characters, Whitespace, double quotes, comma, semicolon, and backslash.

This RegEx isn't either...  It allows way too much.  There is no good in partially filtering the list.  So the feedback is we need something that follows the standards if we are going to do it.

Reply:
RegExp to match field-content in RFC 7230 sec 3.2 
 * field-content = field-vchar [ 1*( SP / HTAB ) field-vchar ]
 * field-vchar   = VCHAR / obs-text
 * obs-text      = %x80-FF
```js
/^[\u0009\u0020-\u007e\u0080-\u00ff]+$/
```